File: Lintcode/algo/lintcode-0609-TwoSumLessThanOrEqualToTarget-[array]-[medium]/Solution.two-pointers.java
public class Solution {
    /**
     * @param nums: an array of integer
     * @param target: an integer
     * @return: an integer
     */
    public int twoSum5(int[] nums, int target) {
        // write your code here
        Arrays.sort(nums);
        int cnt = 0;
        int start = 0, end = nums.length - 1;
        while (start < end) {
            if (nums[start] + nums[end] > target) {
                end--;
            } else {
                cnt += end - start;
                start++;
            }
        }

        return cnt;
    }
}

File: Leetcode/leetcode-0027-RemoveElement-[array]-[easy]/Solution.two-pointers.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function (nums, val) {
  let i = 0,
    j = 0;
  while (i < nums.length) {
    if (nums[i] !== val) {
      nums[j] = nums[i];
      j++;
    }

    i++;
  }

  return j;
};
